### PR TITLE
wallet-cli: allow priority argument for sweep_all and donate

### PR DESCRIPTION
**Pull request by stoffu**

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -3268,6 +3281,18 @@ bool simple_wallet::donate(const std::vector<std::string> &args_)
...
+  {
+    auto priority_pos = std::find(
+      allowed_priority_strings.begin(),
+      allowed_priority_strings.end(),
+      local_args[0]);
+    if(priority_pos != allowed_priority_strings.end()) {
+      priority_str = local_args[0];
+      local_args.erase(local_args.begin());
+    }
+  }
```
Comment (moneromooo-monero) [inline src/simplewallet/simplewallet.cpp]: That looks like it should be a helper function

Comment (stoffu) [reply src/simplewallet/simplewallet.cpp]: OK, now fixed, does it look good?

**Comment by stoffu** [comment]
Wait, I don’t see how you can leave the erase out. What’s subjective?

**Comment by moneromooo-monero** [comment]
I'd have left it out of the function, as it's parsing function. if parsing succeeds, the caller would have removed. One of the callers actually adds code to work around the removal. Choosing to do this instead of what you did is what is subjective.

**Comment by stoffu** [comment]
Ah, now it makes sense to me, and I agree with your idea. Now I do it differently. Do you like it better now?

**Comment by fluffypony** [comment]
Please rebase

**Comment by stoffu** [comment]
@fluffypony  Rebased

**Pull request merged**